source/common/quic/quic_transport_socket_factory.cc

```diff
@@ -16,8 +16,9 @@ QuicServerTransportSocketConfigFactory::createTransportSocketFactory(
       config, context.messageValidationVisitor());
   auto server_config = std::make_unique<Extensions::TransportSockets::Tls::ServerContextConfigImpl>(
       quic_transport.downstream_tls_context(), context);
-  auto factory =
-      std::make_unique<QuicServerTransportSocketFactory>(context.scope(), std::move(server_config));
+  auto factory = std::make_unique<QuicServerTransportSocketFactory>(
+      (quic_transport.has_enable_early_data() ? quic_transport.enable_early_data().value() : true),
```

Comment:
I think this could also be: `!quic_transport.has_enable_early_data() || quic_transport.enable_early_data().value()`